crates/orderbook/src/database/orders.rs

```diff
@@ -90,19 +91,31 @@ pub struct OrderWithQuote {
 }
 
 impl OrderWithQuote {
-    pub fn new(order: Order, quote: Option<Quote>) -> Self {
-        Self {
-            quote: quote.map(|quote| orders::Quote {
-                order_uid: ByteArray(order.metadata.uid.0),
-                gas_amount: quote.data.fee_parameters.gas_amount,
-                gas_price: quote.data.fee_parameters.gas_price,
-                sell_token_price: quote.data.fee_parameters.sell_token_price,
-                sell_amount: u256_to_big_decimal(&quote.sell_amount),
-                buy_amount: u256_to_big_decimal(&quote.buy_amount),
-                solver: ByteArray(quote.data.solver.0),
-            }),
+    pub fn try_new(order: Order, quote: Option<Quote>) -> Result<Self, AddOrderError> {
```

Comment:
Passing by value in case of `try_new()` function means that it takes the ownership of `order` and `quote` and then creates from them `OrderWithQuote`. Also the callers of `try_new()` is not using `order` and `quote` after calling `try_new()` function, so there is no need to clone them in the callers. Changing to passing by reference means that we need to explicitly call `clone()` on some of the fields of `order` and `quote`, so I think pass by value is more clean solution in this case.
From performance perspective I think both options (pass by value and pass by reference) will give same code after compiler optimizations.